DeepSpeech.py

```diff
@@ -257,7 +262,7 @@ def initialize_globals():
 
     # The number of characters in the target language plus one
     global n_character
-    n_character = 29 # TODO: Determine if this should be extended with other punctuation
+    n_character = alphabet.size() + 1 # +1 for CTC blank label
```

Comment:
Note that this means if you change the alphabet file's size checkpoints will fail to be restored. There's no plug and play solution if you want to start training a different language on top of an existing checkpoint, but this helps.

Reply:
Hi @reuben.
I just tested your mods..
Simply added 'é' lign 29 in alphabet.txt

error : 
```
Exception in thread Thread-3:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/nvidia/DeepSpeech/util/feeding.py", line 148, in _populate_batch_queue
    target = text_to_char_array(transcript, self._alphabet)
  File "/home/nvidia/DeepSpeech/util/text.py", line 36, in text_to_char_array
    return np.asarray([alphabet.label_from_string(c) for c in result])
  File "/home/nvidia/DeepSpeech/util/text.py", line 23, in label_from_string
    return self._str_to_label[string]
KeyError: '\xc3'
```

Reply:
my alphabet : 
```
...
x
y
z
'
é
```
a wav sample with the train.csv if you want to test special character : 
http://dl.free.fr/rR2XQjG1o
Ready to test, or help !

Reply:
I replaced the last 'e' in data/ldc93s1/ldc93s1.csv with an 'é', added é to `alphabet.txt`, and it worked. Are you sure those are the only changes you made to the code?

Reply:
oh, I forgot to make !
but error now : 
**`bazel build -c opt --copt=-march=native --copt=-mtune=native --copt=-O3 //tensorflow:libtensorflow_cc.so //native_client:*`**
```
ERROR: /home/nvidia/tensorflow/native_client/BUILD:3:1: file '//tensorflow:tensorflow.bzl' does not contain symbol 'if_linux_x86_64'
ERROR: /home/nvidia/tensorflow/native_client/BUILD:34:18: Traceback (most recent call last):
	File "/home/nvidia/tensorflow/native_client/BUILD", line 17
		cc_library(name = "deepspeech_utils", srcs = ..."], <4 more arguments>)
	File "/home/nvidia/tensorflow/native_client/BUILD", line 34, in cc_library
		if_linux_x86_64
name 'if_linux_x86_64' is not defined
ERROR: error loading package 'native_client': Package 'native_client' contains errors
INFO: Elapsed time: 0.078s
FAILED: Build did NOT complete successfully (1 packages loaded)
    currently loading: tensorflow
```

Reply:
I deleted my Deepspeech dir, pasted your fixed one, 
i did a python setup.py install on native_client,

cd tensorflow
ln -s ../DeepSpeech/native_client ./

but the bazel build... stops with errors ?!!

An idea ?

Reply:
@reuben 
Perhaps i miss anything !
after erasing my deepspeech dir, replaced by your fixed dir, 
I did : bazel build -c opt --copt=-march=native --copt=-mtune=native --copt=-O3 //tensorflow:libtensorflow_cc.so //native_client:*
on tensorflow dir,

cd ../DeepSpeech/native_client
make deepspeech

PREFIX=/usr/local/lib sudo make install

and tested only ldc93s1 dir, with exactly the last é in csv and é in ent alphabet.txt.
Same error !
I didn't touch to original words.txt, neither lm

Reply:
@elpimous Py2 or Py3 ? You should come on IRC to diagnose, would be easier :)

Reply:
Ok thanks

Reply:
Did you re-train the network?

Reply:
Oh, and make sure your LM knows about the word yéar, or disable the LM correction.